### PR TITLE
Swagger: Set strict CSP in /swagger

### PR DESCRIPTION
**Pull request by KristianGrafana**
**What is this feature?**

In response to CVE-2024-6837 - since the `/swagger` page is mainly 3rd party code, and exposed by default, it make sense to add appropriate security measurements to that page. This PR will add a strict CSP to that page, so in the event of a potential XSS in Swagger, it will be properly mitigated. 

Note that this PR does **not** allow the CSP to be turned off for the `/swagger` endpoint. The CSP also contains `require-trusted-types-for 'script'` to help combat any DOM XSS. Full changelog:

* Sets a strict CSP + TT that can not be turned off through configuration in the `/swagger` endpoint, both through the header, and as in a meta tag. 
* The swagger Trusted Types policy for `createScript` returns empty.
* Remove violating `<script></script>` in the swagger page.
* Make the `GenerateNonce()` Go function global.

**Why do we need this feature?**

We should add approach security measurements to code that we do not control, especially when it's exposed by default. We need this feature to be safe from XSS attacks. 

**Who is this feature for?**

Us, and Grafana admins. In the event of an XSS, the CSP will mitigate it and no rush to patch. 

**Special notes for your reviewer:**

I have tested this in Firefox, Chrome and Safari and found no issues, both logged in and unauthenticated. I've also tested with and without CSP (`content_security_policy`) enabled. I have not tested this in HG where we do have a CSP enabled, but will test it once CI spins up an instance. 

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/swagger/index.tsx [hunk 1]
```diff
@@ -28,9 +28,9 @@ import { Page } from './SwaggerPage';
 const tt = window.trustedTypes;
 if (tt?.createPolicy) {
   tt.createPolicy('default', {
-    createHTML: (string, sink) => DOMPurify.sanitize(string, { RETURN_TRUSTED_TYPE: true }) as unknown as string,
-    createScriptURL: (url, sink) => textUtil.sanitizeUrl(url) as unknown as string,
-    createScript: (script, sink) => script,
+    createHTML: (string) => DOMPurify.sanitize(string, { RETURN_TRUSTED_TYPE: true }) as unknown as string,
```
Comment (tskarhed) [inline public/swagger/index.tsx:31]: Is the type casting needed here? We managed to get away without it in Grafana 🤔

File: public/swagger/index.tsx [hunk 2]
```diff
@@ -28,9 +28,9 @@ import { Page } from './SwaggerPage';
...
+    createScriptURL: (url) => textUtil.sanitizeUrl(url) as unknown as string,
+    createScript: () => "",
```
Comment (tskarhed) [inline public/swagger/index.tsx:33]: Not a bad idea. Does this prohibit sending requests from the Swagger UI though?

**Review by ryantxu — APPROVED** [review]
🎉  works as expected -- great to get this out and tested

File: public/swagger/index.tsx (hunk 2 above)
Comment (ryantxu) [reply public/swagger/index.tsx:33]: I tested all the UI features I use (sending requests, etc) and it all seems good

File: public/swagger/index.tsx (hunk 1 above)
Comment (tskarhed) [reply public/swagger/index.tsx:31]: Oh, sorry, we didn't use DOMPurify there

File: public/swagger/index.tsx (hunk 2 above)
Comment (KristianGrafana) [reply public/swagger/index.tsx]: That is controlled by the `connect-src` directive, which is not set. The goal of this PR is to protect against XSS. The trusted types policy will just interfere with using "unsafe" DOM API, and should not disturb any requests from/to Swagger UI.

**Comment by KristianGrafana** [comment]
This works as expected. Monaco Editor is throwing TT violations when creating Workers, but Monaco is fallbacking to main thread so it work as normal nonetheless.

**Pull request closed without merging**